From 25f6ed492a8da150304793afc9658d4acfe834d8 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Fri, 6 Jan 2017 17:43:20 -0800
Subject: [PATCH] wsdd2: Fix crash bug with VPN interfaces

A user had OpenVPN installed, and the virtual network interfaces
created by OpenVPN caused wsdd2 to crash due to some unexpected
attributes.

https://community.netgear.com/t5/Using-your-ReadyNAS/wsdd2-service-failure-after-reboot/m-p/1203764
---
 packages/wsdd2/debian/changelog | 6 ++++++
 packages/wsdd2/wsdd2.c          | 7 ++++++-
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/packages/wsdd2/debian/changelog b/packages/wsdd2/debian/changelog
index 9e6ed553d..c4e98a820 100644
--- a/packages/wsdd2/debian/changelog
+++ b/packages/wsdd2/debian/changelog
@@ -1,3 +1,9 @@
+wsdd2 (1.5) stable; urgency=medium
+
+  * Fix crash bug with VPN interfaces.
+
+ -- Justin Maggard <justin.maggard@netgear.com>  Fri, 06 Jan 2017 17:42:49 -0800
+
 wsdd2 (1.4) unstable; urgency=medium
 
   * Fix some memory leaks.
diff --git a/packages/wsdd2/wsdd2.c b/packages/wsdd2/wsdd2.c
index 3b31b30a2..14121ae0d 100644
--- a/packages/wsdd2/wsdd2.c
+++ b/packages/wsdd2/wsdd2.c
@@ -675,9 +675,14 @@ again:
 
 		if (sv->family == AF_INET || sv->family == AF_INET6) {
 			for (ifa = ifaddrs; ifa; ifa = ifa->ifa_next) {
-				if (ifa->ifa_addr->sa_family != sv->family ||
+				if (!ifa->ifa_addr ||
+					(ifa->ifa_addr->sa_family != sv->family) ||
 					(ifa->ifa_flags & IFF_LOOPBACK) ||
 					(!strcmp(ifa->ifa_name, "LeafNets")) ||
+					(!strncmp(ifa->ifa_name, "docker", 6)) ||
+					(!strncmp(ifa->ifa_name, "veth", 4)) ||
+					(!strncmp(ifa->ifa_name, "tun", 3)) ||
+					(!strncmp(ifa->ifa_name, "zt", 2)) ||
 					(sv->mcast_addr &&
 					!(ifa->ifa_flags & IFF_MULTICAST)))
 					continue;
-- 
2.11.0

